core/domain/stats_domain.py

```diff
@@ -455,7 +588,7 @@ def __str__(self):
             self.__class__.__name__,
             ', '.join('%s=%r' % (prop, getattr(self, prop)) for prop in props))
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
```

Comment:
Why is `object` here?

Reply:
If a specific object such as StateStats is used instead, mypy throws an error that the Liskov substitution principle has been violated and suggests using `object` and `NotImplemented` instead.

Reply:
Done